app/collections/settings.js

```diff
@@ -1,89 +1,91 @@
-const collectionsPath = path.join(userDataPath, 'collections');
-const Settings = new Datastore({
-    filename:              path.join(collectionsPath, 'settings.db'),
-    autoload:              true,
-    corruptAlertThreshold: 1,
-    timestampData:         true
-});
-
-export const saveGradeSystem = data => {
-    Settings.insert(data, error => {
-        if (error) {
-            updateFailed();
-        }
-    });
-};
-
-export const getAddressData = () => new Promise(resolve => Settings.find({}, (err, entry) => {
-    if (err) {
-        unableToRetrieve();
+import connectionToDB from './connectionSetup';
+import {displayToast} from '../notifications';
+
+const Settings = connectionToDB('settings');
+
+export async function saveGradeSystem(data) {
+    try {
+        await Settings.insert(data);
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
     }
+}
+
+export async function getAddressData() {
+    try {
+        const results = await Settings.find({});
 
-    return resolve(entry);
-}));
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
 
-export const getSystemType = () => new Promise(resolve => Settings.find({}, (err, docs) => {
-    if (err) {
-        updateFailed();
+        return null;
     }
+}
 
-    return resolve(docs);
-}));
+export async function getSystemType() {
+    try {
+        const results = await Settings.find({});
+
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
+
+        return null;
+    }
+}
 
-const updateAddress = (previous, id) => {
+async function updateAddress(previous, id) {
     const {title, street, province, country, zip, city, year} = previous;
-    Settings.update(
-        {_id: id},
-        {$set: {title, street, province, country, zip, city, year}},
-        {},
-        err => {
-            if (err) {
-                saveFailed();
-            }
-            saveSuccessful();
-        }
-    );
-};
-
-const updateSystem = (previous, id) => {
+    try {
+        await Settings.update({_id: id}, {$set: {title, street, province, country, zip, city, year}}, {});
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
+    }
+}
+
+async function updateSystem(previous, id) {
     const {note, points, percent} = previous;
+    try {
+        await Settings.update({_id: id}, {$set: {note, points, percent}}, {});
+        displayToast('saveSuccess');
+    } catch (e) {
+        displayToast('saveFail');
+        console.log(e);
+    }
+}
 
-    Settings.update({_id: id}, {$set: {note, points, percent}}, {}, err => {
-        if (err) {
-            saveFailed();
-        }
-        saveSuccessful();
-    });
-};
+export async function updateGradeType(data) {
+    let results = await getSystemType();
+    try {
+        await updateSystem(data, results[0]._id);
```

Comment:
@hcuffy what if  `results[0]` is undefined?

Reply:
fixed